inc/zoo/swar/SWAR.h

```diff
@@ -114,15 +127,21 @@ struct SWAR {
         return SWAR(value() >> (NBits * laneCount));
     }
 
-    constexpr auto blitElement(int index, T value) const noexcept {
-        auto elementMask = ((T(1) << NBits) - 1) << (index * NBits);
-        return SWAR((m_v & ~elementMask) | (value << (index * NBits)));
+    /// \brief as the name suggests
+    /// \param protectiveMask should clear the bits that would cross the lane.
+    /// The bits that will be cleared are directly related to the count of shifts, it is natural to maintain
+    /// the protective mask by the caller, otherwise, the mask will be computed on all invocations.
+    /// Not sure the optimizer would maintain this mask somewhere to not calculate it.
```

Comment:
```suggestion
    /// We are not sure the optimizer would maintain this mask somewhere, if it was to recalculate it it would be disastrous for performance.
```